plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,341 @@
+	"CREATE",
+	"MKDIR",
+	"SYMLINK",
+	"MKNOD",
+	"REMOVE",
+	"RMDIR",
+	"RENAME",
+	"LINK",
+	"READDIR",
+	"READDIRPLUS",
+	"FSSTAT",
+	"FSINFO",
+	"PATHCONF",
+	"COMMIT",
+}
+
+var nfs4Fields = []string{
+	"NULL",
+	"READ",
+	"WRITE",
+	"COMMIT",
+	"OPEN",
+	"OPEN_CONFIRM",
+	"OPEN_NOATTR",
+	"OPEN_DOWNGRADE",
+	"CLOSE",
+	"SETATTR",
+	"FSINFO",
+	"RENEW",
+	"SETCLIENTID",
+	"SETCLIENTID_CONFIRM",
+	"LOCK",
+	"LOCKT",
+	"LOCKU",
+	"ACCESS",
+	"GETATTR",
+	"LOOKUP",
+	"LOOKUP_ROOT",
+	"REMOVE",
+	"RENAME",
+	"LINK",
+	"SYMLINK",
+	"CREATE",
+	"PATHCONF",
+	"STATFS",
+	"READLINK",
+	"READDIR",
+	"SERVER_CAPS",
+	"DELEGRETURN",
+	"GETACL",
+	"SETACL",
+	"FS_LOCATIONS",
+	"RELEASE_LOCKOWNER",
+	"SECINFO",
+	"FSID_PRESENT",
+	"EXCHANGE_ID",
+	"CREATE_SESSION",
+	"DESTROY_SESSION",
+	"SEQUENCE",
+	"GET_LEASE_TIME",
+	"RECLAIM_COMPLETE",
+	"LAYOUTGET",
+	"GETDEVICEINFO",
+	"LAYOUTCOMMIT",
+	"LAYOUTRETURN",
+	"SECINFO_NO_NAME",
+	"TEST_STATEID",
+	"FREE_STATEID",
+	"GETDEVICELIST",
+	"BIND_CONN_TO_SESSION",
+	"DESTROY_CLIENTID",
+	"SEEK",
+	"ALLOCATE",
+	"DEALLOCATE",
+	"LAYOUTSTATS",
+	"CLONE",
+}
+
+var nfsopFields = []string{
+	"ops",
+	"trans",
+	"timeouts",
+	"bytes_sent",
+	"bytes_recv",
+	"queue_time",
+	"response_time",
+	"total_time",
+}
+
+func convert(line []string) []float64 {
+	var nline []float64
+	for _, l := range line[1:] {
+		f, _ := strconv.ParseFloat(l, 64)
+		nline = append(nline, f)
+	}
+	return nline
+}
+
+func In(list []string, val string) bool {
+	for _, v := range list {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *NFSCLIENT) parseStat(mountpoint string, export string, version string, line []string, acc telegraf.Accumulator) error {
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convert(line)
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var fields = make(map[string]interface{})
+
+	if version == "3" || version == "4" {
+		if In(nfs3Fields, first) {
+			if first == "READ" {
+				fields["read_ops"] = nline[0]
+				fields["read_retrans"] = (nline[1] - nline[0])
+				fields["read_bytes"] = (nline[3] + nline[4])
+				fields["read_rtt"] = nline[6]
+				fields["read_exe"] = nline[7]
+				acc.AddFields("nfsstat_read", fields, tags)
+			} else if first == "WRITE" {
+				fields["write_ops"] = nline[0]
+				fields["write_retrans"] = (nline[1] - nline[0])
+				fields["write_bytes"] = (nline[3] + nline[4])
+				fields["write_rtt"] = nline[6]
+				fields["write_exe"] = nline[7]
+				acc.AddFields("nfsstat_write", fields, tags)
+			}
+		}
+	}
+	return nil
+}
+
+func (n *NFSCLIENT) parseData(mountpoint string, export string, version string, line []string, acc telegraf.Accumulator) error {
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convert(line)
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var fields = make(map[string]interface{})
+
+	if first == "events" {
+		for i, t := range eventsFields {
+			fields[t] = nline[i]
```

Comment:
Check that `len(nline)` is at least as long as `eventFields` to avoid a potential panic.